raysort/main_push.py

```diff
@@ -59,16 +63,20 @@ def mapper(
 
 def _merge_blocks(
     blocks: list[np.ndarray], bounds: list[int]
-) -> Iterable[Union[np.ndarray, int]]:
+) -> Iterable[Union[np.ndarray, Tuple, int]]:
     total_bytes = sum(b.size for b in blocks)
     num_records = max(1, constants.bytes_to_records(total_bytes))
     get_block = lambda i, d: blocks[i] if d == 0 else None
     merger = sortlib.merge_partitions(
         len(blocks), get_block, num_records, False, bounds
     )
+
+    datachunk_sizes = []
     for datachunk in merger:
         yield datachunk
-    yield total_bytes
+        datachunk_sizes.append(datachunk.size)
+
+    yield MergeStats(ray.put(tuple(datachunk_sizes)), ray.put(total_bytes))
```

Comment:
No need to `ray.put()`. Does it give you error if you don't use `ray.put()`?

Reply:
If I don't do ray.put ray will only wrap the MergeStats in a object ref, but the two parameters inside aren't object refs. I can remove the ray.put around the datachunk_sizes, but we need the total_bytes to be an object reference so that we can use it as the key in a dictionary later (if total_bytes is an int it might not be unique).